pkg/storage/driver/driver.go

```diff
@@ -29,6 +29,8 @@ var (
 	ErrReleaseNotFound = errors.New("release: not found")
 	// ErrReleaseExists indicates that a release already exists.
 	ErrReleaseExists = errors.New("release: already exists")
+	// ErrReleaseStorageLimit indicates that a release storage limit exceeded, reference: https://github.com/kubernetes/kubernetes/blob/release-1.18/staging/src/k8s.io/apiserver/pkg/server/config.go#L323
+	ErrReleaseStorageLimit = errors.New("release: storage limit is 3MB")
```

Comment:
This is an exported error so we won't be changing the content of it during Helm v3. That would be a breaking change for package consumers via SemVer. With that in mind, I do not like hard coding a value for the storage limit. If (or when) that happens we could run into issues.

Also, the storage size is configurable in etcd to some extent. There are docs at https://etcd.io/docs/v3.4.0/dev-guide/limit/

I'm by no means an etcd expert.

Can we have the error respond in a way that doesn't hard code the storage limit in a Helm public variable?

Reply:
This is a hard code defined by kubernetes: [limits the size of the requested body](https://github.com/kubernetes/kubernetes/blob/release-1.18/staging/src/k8s.io/apiserver/pkg/server/config.go#L323).
Our only dependency is api-server. We can't sense the limit size of ETCD. We should not consider the limit of ETCD. If you don't consider hard coding, you can consider trying to extract the API's error information and parse its limit size.

Reply:
I use `errors.Wrapf(err, "release: storage limit")` to package it, output:
```
Error: release: storage limit: Request entity too large: limit is 3145728
```
The problem of hard coding is solved.There is no explicit prompt to the user, kubernetes does not add units.